server/aws-lsp-codewhisperer/src/language-server/netTransform/transformHandler.ts

```diff
@@ -0,0 +1,255 @@
+            console.log('response start transform api: ' + JSON.stringify(response))
+            return getCWStartTransformResponse(response, uploadId)
+        } catch (error) {
+            const errorMessage = (error as Error).message ?? 'Error in StartTransformation API call'
+            console.log(errorMessage)
+            throw new Error(errorMessage)
+        }
+    }
+
+    async preTransformationUploadCode(userInputrequest: QNetStartTransformRequest): Promise<string> {
+        let uploadId = ''
+        const randomPath = uuidv4().substring(0, 4)
+        const workspacePath = path.join(os.tmpdir(), randomPath)
+        try {
+            const payloadFilePath = await this.zipCodeAsync(userInputrequest, workspacePath)
+            console.log('payload path: ' + payloadFilePath)
+            uploadId = await this.uploadPayloadAsync(payloadFilePath)
+            console.log('artifact successfully uploaded. upload tracking id: ' + uploadId)
+        } catch (error) {
+            const errorMessage = (error as Error).message ?? 'Failed to upload zip file'
+            throw new Error(errorMessage)
+        } finally {
+            cleanup(workspacePath)
+        }
+        return uploadId
+    }
+
+    async uploadPayloadAsync(payloadFileName: string): Promise<string> {
+        const sha256 = getSha256(payloadFileName)
+        let response: CreateUploadUrlResponse
+        try {
+            const apiStartTime = Date.now()
+            response = await this.client.codeModernizerCreateUploadUrl({
+                contentChecksum: sha256,
+                contentChecksumType: 'SHA_256',
+                uploadIntent: 'TRANSFORMATION',
+            })
+        } catch (e: any) {
+            const errorMessage = (e as Error).message ?? 'Error in CreateUploadUrl API call'
+            console.log('Error: ', errorMessage)
+            throw new Error(errorMessage)
+        }
+
+        try {
+            await this.uploadArtifactToS3Async(payloadFileName, response)
+        } catch (e: any) {
+            const errorMessage = (e as Error).message ?? 'Error in uploadArtifactToS3 call'
+            console.log('Error: ', errorMessage)
+            throw new Error(errorMessage)
+        }
+        return response.uploadId
+    }
+
+    async zipCodeAsync(request: QNetStartTransformRequest, basePath: string): Promise<string> {
+        try {
+            return await createZip(request, basePath)
+        } catch (e: any) {
+            console.log({ cause: e as Error })
+        }
+        return ''
+    }
+
+    async uploadArtifactToS3Async(fileName: string, resp: CreateUploadUrlResponse) {
+        const sha256 = getSha256(fileName)
+        const headersObj = this.getHeadersObj(sha256, resp.kmsKeyArn)
+        try {
+            const response = await fetch(resp.uploadUrl, {
+                method: 'PUT',
+                body: fs.readFileSync(fileName),
```

Comment:
@kmile @ege0zcan i need a way to use ReadFileSync, Currently i dont see `workspace.fs.readFileSync` available. It only has `workspace.fs.readFile` read file. Any workaround for that?

Reply:
Why does this need to be a synchronous call?

Wouldn't 

```
const body = await fs.readFile(fileName),
const response = await fetch(resp.uploadUrl, {
                method: 'PUT',
                body: body,
                 headers: headersObj,
            })
```

Have the same effect?

Reply:
This is not working as it its. Its giving me 400 error when i use `workspace.fs.readFile`